crates/sui-types/src/programmable_transaction_builder.rs

```diff
@@ -296,23 +296,32 @@ impl ProgrammableTransactionBuilder {
                 amounts.len()
             )
         }
+        if amounts.is_empty() {
+            return Ok(());
+        }
 
         // collect recipients in the case where they are non-unique in order
         // to minimize the number of transfers that must be performed
         let mut recipient_map = IndexMap::new();
+        let mut amt_args = vec![];
         for (recipient, amount) in recipients.into_iter().zip(amounts) {
+            let cur = amt_args.len();
             recipient_map
                 .entry(recipient)
                 .or_insert_with(Vec::new)
-                .push(amount);
+                .push(cur);
```

Comment:
nit: `enumerate` instead of calculating `cur` + `or_default` instead of `or_insert_with`?

```suggestion
        for (i, (recipient, amount)) in recipients.into_iter().zip(amounts).enumerate() {
            recipient_map
                .entry(recipient)
                .or_default()
                .push(i);
```